### ISSUE TITLE
[Bug] broker jvm crash 

### ISSUE THREAD

**Issue by zcola** [apache/pulsar#21577]
### Search before asking

- [X] I searched in the [issues](https://github.com/apache/pulsar/issues) and found nothing similar.

### Version

2.11.1 

### Minimal reproduce step

![image](https://github.com/apache/pulsar/assets/3234989/c02f8590-0992-4530-9267-aee12c0c297d)
It happened several times randomly and is not easy to reproduce. There are multiple brokers.

```
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f05ac3c07f2, pid=1, tid=652
```

[hs_err_pid1.log](https://github.com/apache/pulsar/files/13346420/hs_err_pid1.log)
[replay_pid1.log](https://github.com/apache/pulsar/files/13346422/replay_pid1.log)
[docker_broker-2023-11-11.log](https://github.com/apache/pulsar/files/13346424/docker_broker-2023-11-11.log)

### What did you expect to see?

broker crash

### What did you see instead?

no crash

### Are you willing to submit a PR?

- [ ] I'm willing to submit a PR!

**Comment by Technoboy-** [comment]
Could you provide the dump file ?

**Comment by lhotari** [comment]
Thanks for reporting. Do you have a chance to upgrade to 2.11.2 ? 2.11.1 runs on 17.0.6 which is rather old. Some JVM crashes were fixed in 17.0.7 . 

Btw. Probably not related, but I'd recommend setting `/sys/kernel/mm/transparent_hugepage/enabled` to `madvise` when running Java on Linux. (`-XX:+UseTransparentHugePages` can be used with `madvise` to opt-in to use THP for the JVM when needed). Alternatively `-XX:+AlwaysPreTouch` jvm startup option could be used to mitigate some issues that THP causes.

**Comment by zcola** [comment]
> Thanks for reporting. Do you have a chance to upgrade to 2.11.2 ? 2.11.1 runs on 17.0.6 which is rather old. Some JVM crashes were fixed in 17.0.7 .
> 
> Btw. Probably not related, but I'd recommend setting `/sys/kernel/mm/transparent_hugepage/enabled` to `madvise` when running Java on Linux. (`-XX:+UseTransparentHugePages` can be used with `madvise` to opt-in to use THP for the JVM when needed). Alternatively `-XX:+AlwaysPreTouch` jvm startup option could be used to mitigate some issues that THP causes.

Thank you for your reply. After upgrading to 2.11.2, it has not appeared again for a month.
